From 55c413627f8545fb5d2b75efb157d30689985d53 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Miro=20Hron=C4=8Dok?= <miro@hroncok.cz>
Date: Tue, 23 Apr 2013 16:17:43 +0300
Subject: [PATCH] Bugfix: configuration wizard crash 2 #1077

---
 lib/Slic3r/GUI/SkeinPanel.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/Slic3r/GUI/SkeinPanel.pm b/lib/Slic3r/GUI/SkeinPanel.pm
index 1bcaa4d..1a822a0 100644
--- a/lib/Slic3r/GUI/SkeinPanel.pm
+++ b/lib/Slic3r/GUI/SkeinPanel.pm
@@ -234,7 +234,9 @@ sub load_config {
     my ($config) = @_;
     
     foreach my $tab (values %{$self->{options_tabs}}) {
-        $tab->set_value($_, $config->$_) for keys %$config;
+        if ($self->{mode} eq 'expert') {
+            $tab->set_value($_, $config->$_) for keys %$config;
+        }
     }
 }
 
-- 
1.8.1.6

