diff --git a/Build.PL b/Build.PL
index b8921a8..70d542c 100644
--- a/Build.PL
+++ b/Build.PL
@@ -28,7 +28,6 @@ my $build = Module::Build->new(
     recommends     => {
         'Class::XSAccessor' => '0',
         'Growl::GNTP' => '0.15',
-        'Net::DBus'   => '0',
         'XML::SAX::ExpatXS' => '0',
         'Wx'          => '0.9901',
     },
diff --git a/lib/Slic3r/GUI.pm b/lib/Slic3r/GUI.pm
index ceb6adb..48dd662 100644
--- a/lib/Slic3r/GUI.pm
+++ b/lib/Slic3r/GUI.pm
@@ -424,16 +424,6 @@ sub notify {
         $self->{growler}->notify(Event => 'SKEIN_DONE', Title => $title, Message => $message)
             if $self->{growler};
     };
-    if (eval 'use Net::DBus; 1') {
-        eval {
-            my $session = Net::DBus->session;
-            my $serv = $session->get_service('org.freedesktop.Notifications');
-            my $notifier = $serv->get_object('/org/freedesktop/Notifications',
-                                             'org.freedesktop.Notifications');
-            $notifier->Notify('Slic3r', 0, $self->{icon}, $title, $message, [], {}, -1);
-            undef $Net::DBus::bus_session;
-        };
-    }
 }
 
 1;
