diff --git a/xs/Build.PL b/xs/Build.PL
index 51d1a01..cc5805b 100644
--- a/xs/Build.PL
+++ b/xs/Build.PL
@@ -219,6 +219,8 @@ print "\n";
 print 'With @INC: ', join(', ', map "\"$_\"", @INC), "\n";
 print 'With @LIBS: ', join(', ', map "\"$_\"", @LIBS), "\n";
 
+push @LIBS, qw(-lpoly2tri);
+
 my $build = Module::Build::WithXSpp->new(
     module_name     => 'Slic3r::XS',
     dist_abstract   => 'XS code for Slic3r',
