From b71bc371110765141da46b1f8b36efafea21f61c Mon Sep 17 00:00:00 2001
From: Alessandro Ranellucci <aar@cpan.org>
Date: Mon, 17 Jun 2013 19:27:08 +0200
Subject: [PATCH] Remove a couple cases of dangerous aliasing, causing crashes.
 #1257

---
 lib/Slic3r/GUI/SkeinPanel.pm | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/lib/Slic3r/GUI/SkeinPanel.pm b/lib/Slic3r/GUI/SkeinPanel.pm
index 6c4842e..7644136 100644
--- a/lib/Slic3r/GUI/SkeinPanel.pm
+++ b/lib/Slic3r/GUI/SkeinPanel.pm
@@ -247,7 +247,9 @@ sub load_config_file {
     $Slic3r::GUI::Settings->{recent}{config_directory} = dirname($file);
     Slic3r::GUI->save_settings;
     $last_config = $file;
-    $_->load_config_file($file) for values %{$self->{options_tabs}};
+    for my $tab (values %{$self->{options_tabs}}) {
+        $tab->load_config_file($file);
+    }
 }
 
 sub load_config {
@@ -265,7 +267,9 @@ sub config_wizard {
     return unless $self->check_unsaved_changes;
     if (my $config = Slic3r::GUI::ConfigWizard->new($self)->run) {
         if ($self->{mode} eq 'expert') {
-            $_->select_default_preset for values %{$self->{options_tabs}};
+            for my $tab (values %{$self->{options_tabs}}) {
+                $tab->select_default_preset;
+            }
         }
         $self->load_config($config);
     }
-- 
1.8.4

