From b725847a5131f5dc7e82a7eeda1783e9287580ea Mon Sep 17 00:00:00 2001
From: Alessandro Ranellucci <aar@cpan.org>
Date: Wed, 3 Apr 2013 19:26:59 +0200
Subject: [PATCH] Bugfix: configuration wizard led to crash with simple mode.
 #1077

---
 lib/Slic3r/GUI/SkeinPanel.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/Slic3r/GUI/SkeinPanel.pm b/lib/Slic3r/GUI/SkeinPanel.pm
index ef74033..1bcaa4d 100644
--- a/lib/Slic3r/GUI/SkeinPanel.pm
+++ b/lib/Slic3r/GUI/SkeinPanel.pm
@@ -243,7 +243,9 @@ sub config_wizard {
 
     return unless $self->check_unsaved_changes;
     if (my $config = Slic3r::GUI::ConfigWizard->new($self)->run) {
-        $_->select_default_preset for values %{$self->{options_tabs}};
+        if ($self->{mode} eq 'expert') {
+            $_->select_default_preset for values %{$self->{options_tabs}};
+        }
         $self->load_config($config);
     }
 }
-- 
1.8.1.6

